application/client.core/src/app/environment/components/sidebar/shell/environment/preset/component.ts

```diff
@@ -124,7 +124,7 @@ export class SidebarAppShellPresetComponent implements OnInit, OnDestroy {
     }
 
     public _ng_onReset() {
-        this.service.presets[1].information = copy(this.service.defaultInformation);
+        this.service.selectedPreset.information = copy(this.service.defaultInformation);
```

Comment:
Direct access and modification of service's properties isn't good idea in general

Reply:
You still have direct access to `service` properties. Basically, it isn't an issue of code, but an issue of logic. 

Let's say you have a restaurant. And you have a menu. You don't mind changing a menu based on the wishes of your customers. Direct access to service properties, means - each customer can open your menu, remove one page from there and put own one. Another way - create box, which each your customer can use to put his wishes there. And you, based on these wishes, will change a menu.

I am talking about some method, like

```
this.service.setSelectedPresetInfo(info)
```

In your case if we are talking about `default` data. Would not be better to use something like:
```
this.service.resetToDefault();
```
instead
```
this.service.selectedPreset.information = copy(this.service.defaultInformation)
```
?